snappy_wrappers/simple_wrapper.py

```diff
@@ -0,0 +1,126 @@
+"""Abstract wrapper for cnvkit.py"""
+
+import os
+import shutil
+import stat
+import tempfile
+import textwrap
+
+from snakemake.shell import shell
+
+__author__ = "Eric Blanc"
+__email__ = "eric.blanc@bih-charite.de"
+
+
+class SimpleWrapper:
+    header = r"""
+        # Pipe everything to log file
+        if [[ -n "{snakemake.log.log}" ]]; then
+            if [[ "$(set +e; tty; set -e)" != "" ]]; then
+                rm -f "{snakemake.log.log}" && mkdir -p $(dirname {snakemake.log.log})
+                exec &> >(tee -a "{snakemake.log.log}" >&2)
+            else
+                rm -f "{snakemake.log.log}" && mkdir -p $(dirname {snakemake.log.log})
+                echo "No tty, logging disabled" >"{snakemake.log.log}"
+            fi
+        fi
+
+        # Compute md5 except when filename ends with .md5
+        compute_md5() {{
+            fn=$1
+            f=$(basename $fn)
+            d=$(dirname $fn)
+            pushd $d 1> /dev/null 2>&1
+            md5sum $f > $f.md5
+            popd 1> /dev/null 2>&1
+        }}
+
+        # Write out information about conda installation.
+        conda list >{snakemake.log.conda_list}
+        conda info >{snakemake.log.conda_info}
+        compute_md5 {snakemake.log.conda_list}
+        compute_md5 {snakemake.log.conda_info}
+
+        # Create temp directory
+        TMPDIR=$(mktemp -d)
+
+        set -x
+
+        # --------------------------------- Start command -----------------------------------------
+    """
+
+    footer = r"""
+        # --------------------------------- End command -------------------------------------------
+
+        set +x
+
+        for fn in {snakemake.output}
+        do
+            if ! [[ $fn =~ \.md5$ ]]
+            then
+                compute_md5 $fn
+            fi
+        done
+    """
+
+    md5_log = r"""
+        f=$(basename {log})
+        d=$(dirname {log})
+        pushd $d 1> /dev/null 2>&1
+        md5sum $f > $f.md5
+        popd 1> /dev/null 2>&1
```

Comment:
pushd and popd are shell specific commands, so we will have to ensure that the chosen shell supports them.  The shell used for snakemake shell invocations defaults to bash, which has these, so this is fine. Just leaving this here as some information.
(Btw, there's also [shell.prefix and shell.suffix](https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#shell-behavior), which may come in handy)